core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.ts

```diff
@@ -77,6 +77,26 @@ export class SavedWorkflowSectionComponent implements OnInit {
     this.router.navigate(['/workflow/new']).then(null);
   }
 
+  /**
+   * duplicate the current workflow. will had a new record in frontend
+   * workflow list and backend database.
+   */
+  public onClickDuplicateWorkflow(workflowToDuplicate: Workflow): void {
+    const newWorkflow = cloneDeep(workflowToDuplicate);
+    newWorkflow.name = workflowToDuplicate.name + '_copy';
+    newWorkflow.creationTime = Date.now();
+    newWorkflow.lastModifiedTime = Date.now();
+    newWorkflow.wid = undefined;
+    this.workflowPersistService.persistWorkflow(newWorkflow)
```

Comment:
I see you are reusing the persisting undefined workflow logic in order to save a new workflow. This is fine and consistent with our previous assumption.

However, as we discuss more the workflow access control, we might need to extract this "create new workflow" logic out of "persist workflow". 

Can you instead create a new backend endpoint, support a single operation: create a new workflow, based on the workflow content? and let's use the new endpoint instead of persist workflow here. In this way, we can introduce workflow access control in the backend.

Reply:
Sure! If I'm understanding it correctly, to implement the backend endpoint, I sort of need to add a new function written in Scala in the file 'WorkflowResource.scala', right?

Reply:
yes!

Reply:
Ok! Thanks for the quick reply!  I think I kind of know how to do it. I will try to explore it tomorrow (my tomorrow) . I might have some quick questions if I'm stuck.

Reply:
Hi! I think I finished the job, main changes are in 'create new backend endpoint...' commit.